api/v1alpha1/extendeddaemonset_types.go

```diff
@@ -77,7 +77,15 @@ type ExtendedDaemonSetSpecStrategyCanary struct {
 	Duration     *metav1.Duration      `json:"duration,omitempty"`
 	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
 	// +listType=set
-	NodeAntiAffinityKeys []string `json:"nodeAntiAffinityKeys,omitempty"`
+	NodeAntiAffinityKeys []string                                      `json:"nodeAntiAffinityKeys,omitempty"`
+	AutoPause            *ExtendedDaemonSetSpecStrategyCanaryAutoPause `json:"autoPause,omitempty"`
+}
+
+// ExtendedDaemonSetSpecStrategyCanaryAutoPause defines the canary deployment AutoPause parameters of the ExtendedDaemonSet
+// +k8s:openapi-gen=true
+type ExtendedDaemonSetSpecStrategyCanaryAutoPause struct {
+	Enabled     *bool  `json:"enabled,omitempty"`
+	MaxRestarts *int32 `json:"maxRestarts,omitempty"`
```

Comment:
could we please document these fields? especially `MaxRestarts`, we should make it clear that it corresponds to the max number of restarts per canary pod, not the sum of restarts of all canary pods